### ISSUE TITLE
qbittorrent-exporter v1.5.2 causing memory leak on qbittorrent 5.0.2

### ISSUE THREAD

**Issue by agrenott** [martabal/qbittorrent-exporter#96]
Hi,

Thanks for your project. 
I ran into an issue which drove me mad for few days though: enabling the exporter on my qbittorrent 5.0.2 instance caused massive memory leaks on the qbittorrent process (NOT the exporter one).

Sadly I couldn't really find how to debug this issue (or provide more useful insights); I've tried playing with a log of settings of qbit (following various memory leaks issues on project's github), until I had the idea to disable the exporter:
![image](https://github.com/user-attachments/assets/4add0994-2ce3-4324-846f-a9742779ee6d)
While this graph is at node level, you can clearly see the typical leaking trend stopping between Dec. 13th and 14th (which is actually when exporter 1.5.3 got automatically deployed and crashed at startup).

With ~300 torrents in qbit, memory was reaching container's limit (8GB) in ~18 hours.

I'm running both containers in a single kubernetes pod, operated by k3s.
The only "exotic" setting I used was `ENABLE_HIGH_CARDINALITY: true`.

**Comment by martabal** [comment]
Hey :wave:, thanks for the report!

I think this is somehow related to #79. Can you try with the docker image `ghcr.io/martabal/qbittorrent-exporter:pr-97` and/or without the tracker feature (you can disable it with the env `DISABLE_TRACKER=true`)

**Comment by agrenott** [comment]
Hi.
So I've tested with `ghcr.io/martabal/qbittorrent-exporter:pr-97` and behavior is the same (causing a leak on qbittorrent process).

But I may have found something interesting: as I said, I use kubernetes, and a single pod with 2 containers (qbittorrent and qbittorrent-exporter). When the pod starts, exporter is usually much faster than qbit, and thus can't connect immediately. Somehow, it manages to connect later (I guess internal retries). And this is where it trigges the memory leak.

If I kill the exporter process once qbit container is initialized, exporter can run without causing the memory leak on qbit! (ran for 24h without leak).

To recap, the leak happens only when exporter is started before qbittorrent is ready to accept connections.

Hope this helps.

**Comment by martabal** [comment]
Oh right, I think I see what's the problem

**Comment by martabal** [comment]
The new release has the fix which *should* fix it. Please re-open this issue if there's still the memory leak. Again, thank you for the report and the explanations :smile:

**Comment by agrenott** [comment]
Hi, I'm trying with v1.5.6 and it looks good so far.
Thanks!
